mpu6050/mpu6050.go

```diff
@@ -1,95 +1,251 @@
-	x = int32(int16((uint16(data[0])<<8)|uint16(data[1]))) * 15625 / 256
-	y = int32(int16((uint16(data[2])<<8)|uint16(data[3]))) * 15625 / 256
-	z = int32(int16((uint16(data[4])<<8)|uint16(data[5]))) * 15625 / 256
-	return
-}
-
-// ReadRotation reads the current rotation from the device and returns it in
-// µ°/s (micro-degrees/sec). This means that if you were to do a complete
+func (d *Dev) Acceleration() (ax, ay, az int32) {
+	const accelOffset = 0
+	ax = int32(convertWord(d.data[accelOffset+0:])) * 15625 / 512 * d.aRange
+	ay = int32(convertWord(d.data[accelOffset+2:])) * 15625 / 512 * d.aRange
+	az = int32(convertWord(d.data[accelOffset+4:])) * 15625 / 512 * d.aRange
+	return ax, ay, az
+}
+
+// Rotations reads the current rotation from the device and returns it in
+// µ°/rad (micro-radians/sec). This means that if you were to do a complete
 // rotation along one axis and while doing so integrate all values over time,
-// you would get a value close to 360000000.
-func (d Device) ReadRotation() (x int32, y int32, z int32) {
-	data := make([]byte, 6)
-	d.bus.ReadRegister(uint8(d.Address), GYRO_XOUT_H, data)
-	// First the value is converted from a pair of bytes to a signed 16-bit
-	// value and then to a signed 32-bit value to avoid integer overflow.
-	// Then the value is scaled to µ°/s (micro-degrees per second).
-	// This is done in the following steps:
-	// 1. Multiply by 250 * 1000_000
-	// 2. Divide by 32768
-	// The following calculation (x * 15625 / 2048 * 1000) is essentially the
-	// same but avoids overflow. First both operations are divided by 16 leading
-	// to multiply by 15625000 and divide by 2048, and then part of the multiply
-	// is done after the divide instead of before.
-	x = int32(int16((uint16(data[0])<<8)|uint16(data[1]))) * 15625 / 2048 * 1000
-	y = int32(int16((uint16(data[2])<<8)|uint16(data[3]))) * 15625 / 2048 * 1000
-	z = int32(int16((uint16(data[4])<<8)|uint16(data[5]))) * 15625 / 2048 * 1000
-	return
-}
-
-// SetClockSource allows the user to configure the clock source.
-func (d Device) SetClockSource(source uint8) error {
-	return d.bus.WriteRegister(uint8(d.Address), PWR_MGMT_1, []uint8{source})
-}
-
-// SetFullScaleGyroRange allows the user to configure the scale range for the gyroscope.
-func (d Device) SetFullScaleGyroRange(rng uint8) error {
-	return d.bus.WriteRegister(uint8(d.Address), GYRO_CONFIG, []uint8{rng})
-}
-
-// SetFullScaleAccelRange allows the user to configure the scale range for the accelerometer.
-func (d Device) SetFullScaleAccelRange(rng uint8) error {
-	return d.bus.WriteRegister(uint8(d.Address), ACCEL_CONFIG, []uint8{rng})
+// you would get a value close to 6.3 radians (360 degrees).
+func (d *Dev) AngularVelocity() (gx, gy, gz int32) {
+	const angvelOffset = 8
+	gx = int32(convertWord(d.data[angvelOffset+0:])) * 2663 / 5000 * d.gRange
+	gy = int32(convertWord(d.data[angvelOffset+2:])) * 2663 / 5000 * d.gRange
+	gz = int32(convertWord(d.data[angvelOffset+4:])) * 2663 / 5000 * d.gRange
```

Comment:
This introduces a division that was not there previously. I recommend changing this to a power-of-two division (that is optimized to a bitshift) and adjusting the multiply to keep the same output.

Most simple microcontrollers (Cortex-M0 etc) do not include a division instruction and have to emulate it in software, which is slow.